### PR TITLE
expression: fix wrong inferred type for sum and avg (#20926)

### LINKED ISSUE #19907: planner: decimal precisions of sum and avg are different from MySQL
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```sql
create table a(a1 decimal(18, 2));
create view va as select sum(a1), avg(a1) from a;
desc va;
```

### 2. What did you expect to see? (Required)

```sql
+---------+---------------+------+-----+---------+-------+
| Field   | Type          | Null | Key | Default | Extra |
+---------+---------------+------+-----+---------+-------+
| sum(a1) | decimal(40,2) | YES  |     | NULL    |       |
| avg(a1) | decimal(22,6) | YES  |     | NULL    |       |
+---------+---------------+------+-----+---------+-------+
```

### 3. What did you see instead (Required)

```sql
+---------+---------------+------+------+---------+-------+
| Field   | Type          | Null | Key  | Default | Extra |
+---------+---------------+------+------+---------+-------+
| sum(a1) | decimal(65,2) | YES  |      | NULL    |       |
| avg(a1) | decimal(65,6) | YES  |      | NULL    |       |
+---------+---------------+------+------+---------+-------+
```

### 4. What is your TiDB version? (Required)

master

### Explanation

In TiDB:
- the precision of SUM is always 65, and decimal precision is original decimal precision + 4
- the precision of AVG is always 65, and decimal precision stays the same

Refer to https://github.com/pingcap/tidb/blob/master/expression/aggregation/base_func.go#L162 and https://github.com/pingcap/tidb/blob/master/expression/aggregation/base_func.go#L139

In MySQL:
- the precision of SUM is original precision + 22, and decimal precision is original decimal precision + 4
- the precision of AVG is original precision + 4, and decimal precision stays the same

Refer to https://github.com/mysql/mysql-server/blob/f8cdce86448a211511e8a039c62580ae16cb96f5/sql/item_sum.cc#L2207

## SIG slack channel

[#sig-exec](https://slack.tidb.io/invite?team=tidb-community&channel=sig-exec&ref=high-performance)

## Score

- 900

## Mentor

* @lzmhhh123

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #20926 to release-4.0

---

### What problem does this PR solve?

Issue Number: close #19907

### What is changed and how it works?

How it Works:

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

Side effects

- Performance regression
    - Consumes more CPU

### Release note 

- fix wrong inferred type for sum and avg.

### CODE REVIEW

File: planner/core/testdata/stats_suite_out.json [hunk 1]
```diff
@@ -0,0 +1,285 @@
+[
```
Comment (wshwsh12) [inline planner/core/testdata/stats_suite_out.json:1]: I think the file should be created.

**Review by ichn-hu — APPROVED** [review]
LGTM

**Review by zz-jason — DISMISSED** [review]

**Comment by qw4990** [comment]
```
[2021-01-27T06:08:32.505Z] import cycle not allowed in test
[2021-01-27T06:08:32.505Z] package github.com/pingcap/tidb/store/tikv (test)
[2021-01-27T06:08:32.505Z] 	imports github.com/pingcap/tidb/store/mockstore/mocktikv
[2021-01-27T06:08:32.505Z] 	imports github.com/pingcap/tidb/expression/aggregation
[2021-01-27T06:08:32.505Z] 	imports github.com/pingcap/br/pkg/utils
[2021-01-27T06:08:32.505Z] 	imports github.com/pingcap/tidb/statistics/handle
[2021-01-27T06:08:32.505Z] 	imports github.com/pingcap/tidb/infoschema
[2021-01-27T06:08:32.505Z] 	imports github.com/pingcap/tidb/store/helper
[2021-01-27T06:08:32.505Z] 	imports github.com/pingcap/tidb/store/tikv
[2021-01-27T06:08:32.505Z] FAIL	github.com/pingcap/tidb/store/tikv [setup failed]
```
PTAL @lzmhhh123

**Comment by qw4990** [comment]
/run-all-tests

**Review by qw4990 — APPROVED** [review]

**Pull request merged**